app/lib/features/activities/widgets/invitation_card.dart

```diff
@@ -99,4 +81,19 @@ class InvitationCard extends ConsumerWidget {
       loading: () => const CircularProgressIndicator(),
     );
   }
+
+  // method for post-process invitation accept
+  void _onInviteAccept(BuildContext ctx, String roomId) async {
+    EasyLoading.show(status: 'Joining', dismissOnTap: false);
+    bool res = await invitation.accept();
+    if (!res) {
+      EasyLoading.dismiss();
+      EasyLoading.showError('Failed to join');
+      return;
+    }
+    EasyLoading.dismiss();
+    if (ctx.mounted) {
+      ctx.goNamed(Routes.chatroom.name, pathParameters: {'roomId': roomId});
```

Comment:
What if the invitation is of space type?

Reply:
Yeah I just tested, it must have a different route. However, it seems difficult knowing it'll be space or room before redirecting...

Reply:
@gnunicorn Any thoughts on this?

Reply:
I suppose we could check with provider via roomId whether its space or room before routing.

Reply:
we indeed do not currently know whether an invite is for a space or room and that is pretty annoying... element seems to be able to distinguish that somehow though ... we might want to open a ticket to investigate that and leave it as is for now?

Reply:
If we can have `roomType` param in [invitation_profile.dart](https://github.com/acterglobal/a3/blob/main/app/lib/common/models/invitation_profile.dart) then it will solve this issue.

Temporary we can use relevant provider to get `roomType` from `roomId` before routing so that it will not misbehave with specific roomType.

Reply:
> If we can have `roomType` param in [invitation_profile.dart](https://github.com/acterglobal/a3/blob/main/app/lib/common/models/invitation_profile.dart) then it will solve this issue.
> 
> Temporary we can use relevant provider to get `roomType` from `roomId` before routing so that it will not misbehave with specific roomType.

I tried using ```maybeSpaceProvider``` and ```maybeRoomProvider``` to return either `Space` or `Convo` to determine roomType but its not working as supposedly should. It's returning in both cases. We can leave redirection for now and open separate ticket as @gnunicorn said.